src/ImageSharp/Formats/Gif/GifEncoderCore.cs

```diff
@@ -336,15 +335,14 @@ private void WriteColorTable<TPixel>(QuantizedFrame<TPixel> image, EndianBinaryW
             var rgb = default(Rgb24);
             using (IManagedByteBuffer colorTable = this.memoryManager.AllocateManagedByteBuffer(colorTableLength))
             {
-                Span<byte> colorTableSpan = colorTable.Span;
+                ref TPixel paletteRef = ref MemoryMarshal.GetReference(image.Palette.AsSpan());
+                ref Rgb24 rgb24Ref = ref Unsafe.As<byte, Rgb24>(ref MemoryMarshal.GetReference(colorTable.Span));
 
                 for (int i = 0; i < pixelCount; i++)
                 {
-                    int offset = i * 3;
-                    image.Palette[i].ToRgb24(ref rgb);
-                    colorTableSpan[offset] = rgb.R;
-                    colorTableSpan[offset + 1] = rgb.G;
-                    colorTableSpan[offset + 2] = rgb.B;
+                    ref TPixel entry = ref Unsafe.Add(ref paletteRef, i);
+                    entry.ToRgb24(ref rgb);
+                    Unsafe.Add(ref rgb24Ref, i);
```

Comment:
You forgot to assign to assign the reference to a value.

Doesn't this break any tests ??? :fearful:
If not, I think we should add them ASAP. Unsafe optimizations could not be carried out safely without proper coverage.

Reply:
I think you might be reading this wrong. I’m getting the expected output

Reply:
Hold on, you’re right! This path is definitely called, how the hell did I produce correct output. 

I’ll update and investigate